### ISSUE TITLE
ngrok >=3.0.0 support

### ISSUE THREAD

**Issue by exception0x876** [totara/totara-docker-dev#215]
Currently, there is a bug with ngrok version >=3.0.0

**To Reproduce**
1. Run tngrok <sitename>
2. See error "ERROR:  unknown shorthand flag: 'b' in -bind-tls=true"

**Expected behavior**
The ngrok is running without errors.

**Workaround**
drop -bind-tls and change single dash to double dash in --host-header in totara-docker-dev/bin/tngrok
Update config.php and change $_SERVER['HTTP_X_ORIGINAL_HOST'] to $_SERVER['HTTP_X_FORWARDED_HOST'] 

**Desktop (please complete the following information):**
 - OS: macOS

**Comment by exception0x876** [comment]
> @exception0x876 - Thanks for posting this. It fixes the error being thrown running tngrok.
> 
> I have found however that, if caching is enabled, JS will load under the local host instead of the ngrok host.
> 
> Did you run into this?

I haven't tried it with caching enabled. Would you mind posting the relevant cache settings from your config.php?

**Comment by scq** [comment]
I had to also set a few other config options in config.php to make everything work properly and have the generated URLs etc be correct:

```php
if (isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && isset($_SERVER['HTTP_X_FORWARDED_HOST'])) {
    // ngrok 3.x
    $CFG->wwwroot = $_SERVER['HTTP_X_FORWARDED_PROTO'] . '://' . $_SERVER['HTTP_X_FORWARDED_HOST'];
    $CFG->reverseproxy = true;
    if ($_SERVER['HTTP_X_FORWARDED_PROTO'] === 'https') {
        $CFG->sslproxy = true;
    }
}
```

(was a quick fix, so I just added this block after the existing checks)

Will make a PR to add something like this to the default config.php.

**Comment by derschatta** [comment]
With the workaround explained in the description

"Update config.php and change $_SERVER['HTTP_X_ORIGINAL_HOST'] to $_SERVER['HTTP_X_FORWARDED_HOST']"

it worked for me. Created a PR.

@scq did the workaround not work out of the box for you? What happens without the changes you mention?

**Comment by scq** [comment]
@derschatta Ah, I wasn't using tngrok and was passing some different configuration flags. With the same flags as tngrok, that change does work for me.

**Comment by derschatta** [comment]
fixed in #225
